dropwizard-configuration/src/main/java/io/dropwizard/configuration/JsonConfigurationFactory.java

```diff
@@ -0,0 +1,232 @@
+package io.dropwizard.configuration;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.fasterxml.jackson.databind.node.TreeTraversingParser;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A factory class for loading JSON configuration files, binding them to configuration objects, and
+ * validating their constraints. Allows for overriding configuration parameters from system properties.
+ *
+ * @param <T> the type of the configuration objects to produce
+ */
+public class JsonConfigurationFactory<T> implements ConfigurationFactory<T> {
```

Comment:
Very good! But I think we could avoid duplication by extending `JsonConfigurationFactory` from `YamlConfigurationFactory`. We could expose the `getParser` method in `YamlConfigurationFactory`. The method will create a parser from an input stream. `JsonConfigurationFactory` can override this method to produce a `JsonParser` instead of a `YAMLParser`. I had something like that in mind:
```java
public class JsonConfigurationFactory<T> extends YamlConfigurationFactory<T> {

    private final JsonFactory jsonFactory = new JsonFactory();

    public JsonConfigurationFactory(Class<T> klass,
                                    Validator validator,
                                    ObjectMapper objectMapper,
                                    String propertyPrefix) {
        super(klass, validator, objectMapper, propertyPrefix);
    }

    @Override
    protected JsonParser getParser(InputStream input) throws IOException {
        return jsonFactory.createParser(input);
    }

    @Override
    public T build(ConfigurationSourceProvider provider, String path) throws IOException, ConfigurationException {
        try {
            return super.build(provider, path);
        } catch (JsonParseException e) {
            throw ConfigurationParsingException
                .builder("Malformed JSON")
                .setCause(e)
                .setLocation(e.getLocation())
                .setDetail(e.getMessage())
                .build(path);
        }
    }
}
```

YamlConfigurationFactory
```java
protected JsonParser getParser(InputStream input) throws IOException {
        return yamlFactory.createParser(input);
    }

    @Override
    public T build(ConfigurationSourceProvider provider, String path) throws IOException, ConfigurationException {
        try (InputStream input = provider.open(requireNonNull(path))) {
            final JsonNode node = mapper.readTree(getParser(input));
        //...
}
```